modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/DeploymentDirectoryWatcher.java

```diff
@@ -198,12 +211,14 @@ public void stop() {
     stopAppDirMonitorTimer();
 
     deploymentLock.lock();
-    try {
-      notifyStopListeners();
-      stopArtifacts(applications);
-      stopArtifacts(domains);
-    } finally {
-      deploymentLock.unlock();
+    if (disposeArtifactsOnStop) {
+      try {
+        notifyStopListeners();
+        stopArtifacts(applications);
+        stopArtifacts(domains);
+      } finally {
+        deploymentLock.unlock();
```

Comment:
how is this lock released if `disposeArtifactsOnStop == false`

Reply:
A mistake. It should be inside the condition. 
Minor modification: I've changed the method notifyStopListeners so that it is clear what it does (nothing related to listeners but avoiding to persist the status of the app for not starting those apps in case of mule restart).